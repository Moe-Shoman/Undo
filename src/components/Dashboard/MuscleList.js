import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import { Link, Route, BrowserRouter as Router } from 'react-router-dom';
import MuscleDetail from './MuscleDetail';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { doSearch, setMuscle } from '../../actions';
const style = {
  margin: 12,
};

function mapStateToProps(state) {
  return {
    list: state.list,
    selectedMuscle: state.selectedMuscle
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ doSearch, setMuscle }, dispatch)
}

class MuscleList extends React.Component{
  render() {
    return (
      <Router>
        <div>
          <ul>
            <li>
              <Link to='/routines/Chest'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Chest');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Chest" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Shoulders'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Shoulders');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Shoulders" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Traps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Traps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Traps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Quadriceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Quadriceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Quadriceps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Triceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Triceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Triceps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Lats'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Lats');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Lats" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Abdominals'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Abdominals');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Abdominals" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Abductors'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Abductors');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Abductors" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Lower-Back'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Lower Back');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Lower Back" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Adductors'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Adductors');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Adductors" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Middle-Back'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Middle Back');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Middle Back" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Biceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Biceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Biceps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Neck'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Neck');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Neck" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Calves'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Calves');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Calves" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Forearms'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Forearms');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Forearms" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Glutes'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Glutes');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Glutes" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Hamstrings'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Hamstrings');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Hamstrings" style={style} />
              </Link>
            </li>
          </ul>

          <Route path='/routines/:muscle_type' component={MuscleDetail}/>
          <Route exact path='/routines' render={() => (
            <h3>Please pick a muscle group!</h3>
          )}/>
        </div>
      </Router>
    )
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(MuscleList);
