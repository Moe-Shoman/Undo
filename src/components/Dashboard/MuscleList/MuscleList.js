import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import { Link, Route, BrowserRouter as Router } from 'react-router-dom';
import ExerciseList from '../ExerciseList/ExerciseList.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { doSearch } from '../../../actions';
import './MuscleList.css'
const style = {
  margin: 12,
};

function mapStateToProps(state) {
  return {
    list: state.list,
    selectedExercise: state.selectedExercise,
    selectedVideo: state.selectedVideo
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ doSearch }, dispatch)
}

class MuscleList extends React.Component{
  render() {
    console.log("The selected exercise!!!!", this.props.selectedExercise)
    console.log('The selected Video!!!!', this.props.selectedVideo)
    return (
      <Router>
        <div>
          <Route exact path='/routines' render={() => (
            <h3>Please pick a muscle group!</h3>
          )}/>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Chest'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Chest');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Chest" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Shoulders'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Shoulders');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Shoulders" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Traps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Traps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Traps" style={style} />
              </Link>
            </li>
          </ul>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Quadriceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Quadriceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Quadriceps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Triceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Triceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Triceps" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Lats'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Lats');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Lats" style={style} />
              </Link>
            </li>
          </ul>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Abdominals'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Abdominals');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Abdominals" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Abductors'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Abductors');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Abductors" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Lower-Back'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Lower Back');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Lower Back" style={style} />
              </Link>
            </li>
          </ul>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Adductors'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Adductors');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Adductors" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Middle-Back'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Middle Back');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Mid Back" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Biceps'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Biceps');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Biceps" style={style} />
              </Link>
            </li>
          </ul>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Neck'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Neck');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Neck" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Calves'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Calves');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Calves" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Forearms'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Forearms');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Forearms" style={style} />
              </Link>
            </li>
          </ul>
          <ul className='list-group col-md-2 MuscleList-list'>
            <li>
              <Link to='/routines/Glutes'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Glutes');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Glutes" style={style} />
              </Link>
            </li>
            <li>
              <Link to='/routines/Hamstrings'>
                <RaisedButton onClick={(event) => {
                  this.props.doSearch('Hamstrings');
                }} backgroundColor='#C0C0C0' labelColor='#000000' label="Hamstrings" style={style} />
              </Link>
            </li>
          </ul>

          <Route path='/routines/:muscle_type' component={ExerciseList}/>

        </div>
      </Router>
    )
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(MuscleList);
