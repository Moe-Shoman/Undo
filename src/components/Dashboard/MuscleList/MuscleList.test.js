import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MuscleList } from './MuscleList';
import toJson from 'enzyme-to-json';

describe('MuscleList component', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MuscleList>MuscleList</MuscleList>, div);
  });

  test('snapshots', () => {
    const component = shallow(
      <MuscleList>MuscleList</MuscleList>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
