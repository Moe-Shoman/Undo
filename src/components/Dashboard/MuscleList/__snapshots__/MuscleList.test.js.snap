exports[`MuscleList component snapshots 1`] = `
<BrowserRouter>
  <div>
    <Route
      exact={true}
      path="/routines"
      render={[Function]} />
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Chest">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Chest"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Shoulders">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Shoulders"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Traps">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Traps"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
    </ul>
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Quadriceps">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Quadriceps"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Triceps">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Triceps"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Lats">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Lats"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
    </ul>
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Abdominals">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Abdominals"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Abductors">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Abductors"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Lower-Back">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Lower Back"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
    </ul>
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Adductors">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Adductors"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Middle-Back">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Mid Back"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Biceps">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Biceps"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
    </ul>
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Neck">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Neck"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Calves">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Calves"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Forearms">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Forearms"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
    </ul>
    <ul
      className="list-group col-md-2 MuscleList-list">
      <li>
        <Link
          replace={false}
          to="/routines/Glutes">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Glutes"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/routines/Hamstrings">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="Hamstrings"
            labelColor="#000000"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </Link>
      </li>
      <li>
        <a
          href="https://www.weather.com">
          <RaisedButton
            backgroundColor="#C0C0C0"
            disabled={false}
            fullWidth={false}
            label="~*@!Weather!@*~"
            labelColor="#000000"
            labelPosition="after"
            primary={false}
            secondary={false}
            style={
              Object {
                "margin": 12,
              }
            } />
        </a>
      </li>
    </ul>
    <Route
      component={[Function]}
      path="/routines/:muscle_type" />
  </div>
</BrowserRouter>
`;
